### DISCUSSION TITLE
Custom cleaning with  Dreame L10 Prime

### DISCUSSION THREAD

**Discussion by calcioscacchi** [Tasshack/dreame-vacuum#380] [Q&A] [answered]
Hi there. 

I've integrated the robot with success via beta 2.0.0b11 version.
I'm trying to set 2 custom cleaning automations for each room, one is just for sweeping, one is for sweeping and mopping. 

In order to do this, I've tried building a script to call with Alexa, the script is a call to "dreame_vacuum.vacuum_set_custom_cleaning" service.

This is the code for the sweeping one:

`service: dreame_vacuum.vacuum_set_custom_cleaning
target:
  entity_id: vacuum.battista
data:
  segment_id:
    - 7
  suction_level:
    - 2
  repeats:
    - 1
  water_volume:
    - 1
`

The problem is that HomeAssistant gives me the following error: 

"Cannot edit customized cleaning parameters on current map".
Any idea to fix this problem?

Thanks.

**Comment by Tasshack** [comment]
This happens when robot creates a map and you don't save it to the map list or of the saved maps is not currently selected.
Is the app allows you to set customized cleaning parameters when integration does not?

**Reply by calcioscacchi** [reply]
yes it does. I use the dreamehome app and it allows me to select different cleaning modes for each room.

**Comment by Tasshack** [comment, marked as answer]
[Converted to a issue.](https://github.com/Tasshack/dreame-vacuum/issues/381)
